### ISSUE TITLE
Start-restart replication problem in CBL 1.1.0

### ISSUE THREAD

**Issue by ryuzmukhametov** [couchbase/couchbase-lite-ios#696]
Hi,

Recently I stumbled one interesting issue in Couchbase Lite framework.
If after starting a replication I try restart it then the replication is stopped.
I got version 1.1.0 (f1aa2cd07433900d8eebc29f8eb8fe1a81df94ca) with ForestDB support.

For reproduce it you can CBL iOS Test App, and in method updateSyncURL change:
[_pull start];
[_push start];
to:
[_pull start];
[_pull restart];
[_push start];
[_push restart];

After these change my databases replications are stopped.

**Comment by snej** [comment]
I know the example above is simplified to make it easy to describe — have you run into this problem in a real-world situation? The -restart method is literally exactly the same as calling -stop and then -start, so I'm wondering if the problem only occurs if you've just started the replication, or if it'll happen at other times too.

**Comment by pasin** [comment]
I think this is a regression bug. In https://github.com/couchbase/couchbase-lite-ios/commit/5438f4727a9713100c1e5814774687042acc272c, I removed line 312 and 313. I thought those 2 lines are redundant as those will be executed again asynchronously when the -updateStatus: method is called.

Basically when the start method is called right after the stop method, the _started doesn't get reset immediately.

**Comment by ryuzmukhametov** [comment]
Regarding how I have got this bug.
In my app I have method 'actualizeAllReplications' which runs by some application notifications.
In this method I check a replication with my server. If I don't have such replication I create it and start.
If I already have this server replication then I restart it. It happened that 'actualizeAllReplications' was called twice immediately from main thread by notification.
So I got the described situation
